### PR TITLE
feat(internal): Pass 8 CSPRNG random integer words

### PR DESCRIPTION
**Pull request by craciunoiuc**
### Prerequisite checklist

  - [ ] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [ ] Tested your changes against relevant architectures and platforms;
  - [ ] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### CODE REVIEW

File: unikraft/export/v0/ukrandom/params.go
Comment (nderjung) [inline unikraft/export/v0/ukrandom/params.go]: Let's call this var, to reflect its name in the cmdline, just `ParamRandomSeed`.

Subsequent changes are then necessary to the constructor.

**Review by nderjung — APPROVED** [review]
All good on my side implementation-wise. Thanks!

Regarding the interaction with Unikraft itself, I'd like to cc @razvand and @michpappas before merging to help verify everything in conjunction with https://github.com/unikraft/unikraft/pull/1496

**Comment by craciunoiuc** [comment]
Updated to check for RDRAND/RDSEED

If they are both present then ignore the config option and let Unikraft take its own bytes from the device/generator

File: unikraft/export/v0/ukrandom/params.go
Comment (michpappas) [inline unikraft/export/v0/ukrandom/params.go]: You need to prepend each number with `0x` otherwise they won't be interpreted by `uklibparam` neither as valid hex, nor as decimals.

File: internal/cli/kraft/run/runner_kraftfile_runtime.go
Comment (michpappas) [inline internal/cli/kraft/run/runner_kraftfile_runtime.go]: There is a theoretical case where a CPU has `RDSEED` but not `RDRAND`, but I haven't come across such a thing in practice, so I'm not sure if it's worth adding a check. Saying that ,the opposite is possible: `RDRAND` was introduced before `RDSEED`, so there are CPUs out there that have the former but not the latter. The implementation in Unikraft has the minimum requirement of  `RDRAND` when the driver is enabled (seeding functions fall back to `RDRAND`), so in theory you may skip the check for `RDSEED` altogether.

Another thing related to portability: I had a quick look at https://github.com/klauspost/cpuid and it looks like it lacks detection for `FEAT_RNG` on Arm (along with many others). You may consider opening a PR upstream to add these so that they're there by the time we need them.

**Review by michpappas — CHANGES_REQUESTED** [review]
@craciunoiuc just a few comments. Also if you update this, it would be great if you posted a few of the generated cmdlines.

File: internal/cli/kraft/run/runner_kraftfile_runtime.go
Comment (craciunoiuc) [reply internal/cli/kraft/run/runner_kraftfile_runtime.go]: noted, will do

**Comment by craciunoiuc** [comment]
```
qemu-system-x86_64 -append random.seed=[ 0xef59fa23 0x1eba1280 0xb0972ca2 0xc734f02f 0x2079dc04 0x158e0490 0x656d5095 0xa407c115 ] -- /usr/bin/nginx  -cpu qemu64,+pdpe1gb,+rdrand,+rdseed,-vmx,-svm -daemonize -device virtio-net-pci,mac=02:b0:b0:84:3e:01,netdev=hostnet0 -device pvpanic -display none -kernel /home/cez/.local/share/kraftkit/runtime/99876f61-99a0-4072-a2f8-5b6483635e20/unikraft/bin/kernel -machine pc -m size=128M -monitor unix:/home/cez/.local/share/kraftkit/runtime/99876f61-99a0-4072-a2f8-5b6483635e20/qemu_mon.sock,server,nowait -name 99876f61-99a0-4072-a2f8-5b6483635e20 -netdev user,id=hostnet0,hostfwd=tcp::8081-:80 -nographic -no-reboot -S -parallel none -pidfile /home/cez/.local/share/kraftkit/runtime/99876f61-99a0-4072-a2f8-5b6483635e20/machine.pid -qmp unix:/home/cez/.local/share/kraftkit/runtime/99876f61-99a0-4072-a2f8-5b6483635e20/qemu_control.sock,server,nowait -qmp unix:/home/cez/.local/share/kraftkit/runtime/99876f61-99a0-4072-a2f8-5b6483635e20/qemu_events.sock,server,nowait -rtc base=utc -serial file:/home/cez/.local/share/kraftkit/runtime/99876f61-99a0-4072-a2f8-5b6483635e20/machine.log -smp cpus=1,threads=1,sockets=1 -vga none
```

this is how it looks for one run, note that quotes don't show up in the logs

**Review by michpappas — APPROVED** [review]
Looks good, thanks a lot!

Reviewed-by: Michalis Pappas <michalis@unikraft.io>

**Review by nderjung — APPROVED** [review]
Thanks everyone! Looks good from my side. I'll merge this when https://github.com/unikraft/unikraft/pull/1496 is merged into staging.

**Comment by nderjung** [comment]
Linking for completeness: https://github.com/klauspost/cpuid/pull/152

File: internal/cli/kraft/run/runner_kraftfile_runtime.go
Comment (michpappas) [inline internal/cli/kraft/run/runner_kraftfile_runtime.go]: The 1st condition here looks incorrect. It should check if `CONFIG_LIBUKRANDOM_LCPU` is *not* enabled.

File: internal/cli/kraft/run/runner_kraftfile_unikraft.go
Comment (michpappas) [inline internal/cli/kraft/run/runner_kraftfile_unikraft.go]: This one looks like different logic altoghether 🤔

File: internal/cli/kraft/run/runner_package.go
Comment (michpappas) [inline internal/cli/kraft/run/runner_package.go]: This one too should check if `CONFIG_LIBUKRANDOM_LCPU` is *not* enabled.

File: machine/firecracker/v1alpha1.go
Comment (michpappas) [inline machine/firecracker/v1alpha1.go]: Since you're updating this line, the term normally used is "randomness", not "randomization". Or if you want to be more precise, "CPU-generated randomness".

File: machine/qemu/v1alpha1.go
Comment (michpappas) [inline machine/qemu/v1alpha1.go]: Same comment about the message, as above.

**Review by michpappas — CHANGES_REQUESTED** [review]
@craciunoiuc besides the comments below, I think we should have the conditions on for generating the cmdline rewritten in a more clear / readable way, especially given the security-related nature of that code 🙏🏼 

Also since your changes for `FEAT_RNG` were merged upstream, please consider if you should also update the CPUID check for Arm too.

File: internal/cli/kraft/run/runner_kraftfile_unikraft.go
Comment (craciunoiuc) [reply internal/cli/kraft/run/runner_kraftfile_unikraft.go]: did some logic toke reduce the number of negations, will simplfy in another way

**Comment by craciunoiuc** [comment]
Checks are automatically done when checking for the RDRAND features. Imagine it as a an interface with different implementations depending on the OS running the library. So all is good.

File: internal/cli/kraft/run/runner_kraftfile_runtime.go
Comment (michpappas) [inline internal/cli/kraft/run/runner_kraftfile_runtime.go:361]: These are merely "I would had done it differently" / in  `unikraft` we do it differently,  so totally up to you to take into account or ignore. I only care about functional correctness, and the current implementation looks good in that aspect :)

1. I would had written this a bit differently, having a separate variable for the presence of `CONFIG_LIBUKRANDOM`, and then do the check for that at the final condition:
```go
if libUKRandom && hasNoCpuRandomnessSupport && hasCmdlineSupport {
```
 The reason is that there there is no stronger correlation between `CONFIG_LIBUKRANDOM` and `CONFIG_LIBUKRANDOM_CMDLINE_INIT` than `CONFIG_LIBUKRANDOM_CMDLINE_LCPU`.
 
2. I would had also moved this logic into a function to avoid copying (and keeping up to date) into three different places.

3. Looking at the individual commits, it seems that some changes are introduced in one commit just to be updated (read corrected) in another, which people may frown up.

File: machine/firecracker/v1alpha1.go
Comment (michpappas) [inline machine/firecracker/v1alpha1.go]: Should these be additionally conditional to `!hasNoCpuRandomnessSupport`?

File: internal/cli/kraft/run/runner_kraftfile_runtime.go
Comment (craciunoiuc) [reply internal/cli/kraft/run/runner_kraftfile_runtime.go:361]: 1. I can do it, sure
2. If I do it I need to do it in other places too, so I'll skip this one 😆 
3. Yeah, I was a bit split on this also, but I think the bump should be done separately as it has significant changes

File: machine/firecracker/v1alpha1.go
Comment (craciunoiuc) [reply machine/firecracker/v1alpha1.go]: Makes sense

Comment (craciunoiuc) [reply machine/firecracker/v1alpha1.go]: Actually, symbols are not accessible in there, I will move the warning closer to that

**Review by michpappas — APPROVED** [review]
Looks good, thanks!

Reviewed-by: Michalis Pappas <michalis@unikraft.io>

**Review by razvand — APPROVED** [review]
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request merged**
